### PR TITLE
Upload config file, environment file, Dockerfile file

### PR DESCRIPTION
**Pull request by resdenia**

### CODE REVIEW

**Review by yotamloe — CHANGES_REQUESTED** [review]
Looks good, few notes:
* Please add draft `README.md` for Nico
* The otel receiver has more configuration options. I think we should add at least `excluded_images` option for our users. [ref](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/dockerstatsreceiver#configuration)
* A question: This project will handle more modules in the future? Or its designed only for docker metrics?

File: README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+```sh
+git clone https://github.com/logzio/otel-docker-collector-metrics.git otel-docker-collector-metrics
+```
+
+2.  `Docker` to build a container from this image:
+
+```sh
+docker build -t 'otel-docker-collector-metrics:latest' ./otel-docker-collector-metrics
+```
+
```
Comment (mirii1994) [inline README.md:18]: Please remove this section. User should use this image without cloning the repo, just by pulling the docker image and then run it.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+docker run --name otel-docker-collector-metrics \
+ --env METRICS_TOKEN="<<TOKEN>>" \
+ --env DOCKER_ENDPOINT="<<DOCKER_ENDPOINT>>" \
+ --env LOGZIO_LISTENER="<<LOGZIO_LISTENER>>" \
+ --env COLLECTION_INTERVAL="<<COLLECTION_INTERVAL>>" \
+ --env TIMEOUT="<<TIMEOUT>>" \
+ -v /var/run/docker.sock:/var/run/docker.sock \
+  -v $(pwd)/config.yaml:/config.yaml \
+	/otel-docker-collector-metrics:latest
+```
```
Comment (mirii1994) [inline README.md:42]: Remove this part, user should use it by pulling the image, not cloning & building locally :)

File: README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+### Check Logz.io for your metrics
+
+Give your metrics a few minutes to get from your system to ours,
+and then open [Logz.io](https://app.logz.io/#/dashboard/kibana).
+
+You can view your metrics in Grafana.
+We offer preconfigured dashboards for several sources,
+which you can find by clicking **<i class="fas fa-th-large"></i> > Manage**
+in the left menu.
+
```
Comment (mirii1994) [inline README.md:81]: Please add a changelog section

**Review by mirii1994 — CHANGES_REQUESTED** [review]
Great work, please see my comments.
Please have @nico-shishkin go over the documentation.

File: README.md (hunk 3 above)
Comment (resdenia) [reply README.md:52]: Added Versions

File: README.md (hunk 1 above)
Comment (resdenia) [reply README.md]: Removed

**Review by nico-shishkin — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,51 +1,18 @@
-# Opentelementry Docker metrics collector
+# OpenTelementry Docker Metrics Collector
 
 To simplify shipping metrics from one or many sources, we created Docker Metrics Collector. Docker Metrics Collector is a container that runs Metricbeat with the modules you enable at runtime.
```
Comment (Doron-Bargo) [inline README.md:3]: What is the connection for metricbeat?

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 5]
```diff
@@ -1 +1,2 @@
 app.py
```
Comment (mirii1994) [inline .gitignore:1]: I think there isn't an "app.py" file in this repo, so maybe remove it?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,22 @@
+FROM python:3.9-slim
+
+USER root
+
+WORKDIR /
+
+COPY config.yaml ./
+COPY requirements.txt ./
+COPY app.py ./
```
Comment (mirii1994) [inline Dockerfile:9]: Did not see `app.py` in the repo.

File: Dockerfile-arm [hunk 7]
```diff
@@ -0,0 +1,22 @@
+FROM python:3.9-slim
+
+USER root
+
+WORKDIR /
+
+COPY ./config.yaml ./
+COPY requirements.txt ./
+COPY app.py ./
```
Comment (mirii1994) [inline Dockerfile-arm:9]: Did not see app.py in the repo.

**Review by mirii1994 — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -79,6 +46,13 @@ We offer preconfigured dashboards for several sources,
 which you can find by clicking **<i class="fas fa-th-large"></i> > Manage**
 in the left menu.
```
Comment (Doron-Bargo) [inline README.md:47]: Please remove this entire paragraph : 
"...Give your metrics a few minutes to get from your system to ours,
and then open [Logz.io](https://app.logz.io/#/dashboard/kibana).
You can view your metrics in Grafana.
We offer preconfigured dashboards for several sources,
which you can find by clicking **<i class="fas fa-th-large"></i> > Manage**
in the left menu...."
and replace it with : 
"Give your data some time to get from your system to ours, then log in to your Logz.io Metrics account, and open the Logz.io Metrics tab ( and link to https://app.logz.io/#/dashboard/metrics/).

File: README.md [hunk 9]
```diff
@@ -1,51 +1,18 @@
...
-```
-
-or if it's image from Docker hub
+## Run the OpenTelementry Docker Metrics Collector
 
 ```sh
 docker run --name otel-docker-collector-metrics \
- --env METRICS_TOKEN="<<METRICS_TOKEN>>" \
+ --env METRICS_TOKEN="<<METRICS-SHIPPING-TOKEN>>" \
  --env DOCKER_ENDPOINT="<<DOCKER_ENDPOINT>>" \
```
Comment (Doron-Bargo) [inline README.md:16]: This parameter should be optional (with default unix:///var/run/docker.sock ) and therefore not listed here. Also the python code should check if it the user use it and update accordingly

File: README.md [hunk 10]
```diff
@@ -1,51 +1,18 @@
...
  --env LOGZIO_LISTENER="<<LOGZIO_LISTENER>>" \
  --env COLLECTION_INTERVAL="<<COLLECTION_INTERVAL>>" \
```
Comment (Doron-Bargo) [inline README.md:18]: This parameter should be optional (with default 30s) and therefore not listed here. Also the python code should check if it the user use it and update accordingly

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]
Hi , please see my comments

File: config.yaml [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    endpoint: ${LOGZIO_LISTENER}
+    headers:
+      Authorization: Bearer ${METRICS_TOKEN}
+receivers:
+  docker_stats:
+    api_version: 1.24
+    collection_interval: ${COLLECTION_INTERVAL}
+    endpoint: ${DOCKER_ENDPOINT}
+    excluded_images:
+    - pubsub
```
Comment (Doron-Bargo) [inline config.yaml:12]: why we have this section and these images

File: README.md [hunk 12]
```diff
@@ -79,6 +46,13 @@ We offer preconfigured dashboards for several sources,
 which you can find by clicking **<i class="fas fa-th-large"></i> > Manage**
 in the left menu.
 
+## Versions
```
Comment (Doron-Bargo) [inline README.md:49]: Also please state how should the metrics will look like so it will be easy to look for them

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]

File: README.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+
+```sh
+docker run --name otel-docker-collector-metrics \
+ --env METRICS_TOKEN="<<METRICS-SHIPPING-TOKEN>>" \
+ --env DOCKER_ENDPOINT="<<DOCKER_ENDPOINT>>" \
+ --env LOGZIO_LISTENER="<<LOGZIO_LISTENER>>" \
+ --env COLLECTION_INTERVAL="<<COLLECTION_INTERVAL>>" \
+ --env TIMEOUT="<<TIMEOUT>>" \
+ -v /var/run/docker.sock:/var/run/docker.sock \
+ -v $(pwd)/config.yaml:/config.yaml \
```
Comment (mirii1994) [inline README.md:21]: Why do we need this: `-v $(pwd)/config.yaml:/config.yaml`?
The docker file copies it as default, only if someone wants to mount its own config he should use this command, I don't think it should be in the command

File: README.md [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+ logzio/otel-docker-collector-metrics:latest
```
Comment (mirii1994) [inline README.md:22]: The name in this command does not match the name from the docker pull command.

**Review by mirii1994 — CHANGES_REQUESTED** [review]
When I tried to run the current image got this error:
```sh
Traceback (most recent call last):
  File "//app.py", line 36, in <module>
    w.store_array_in_yaml()
  File "//app.py", line 14, in store_array_in_yaml
    config('EXCLUDED_IMAGES'))
  File "/usr/local/lib/python3.9/site-packages/decouple.py", line 197, in __call__
    return self.config(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/decouple.py", line 85, in __call__
    return self.get(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/decouple.py", line 70, in get
    raise UndefinedValueError('{} not found. Declare it as envvar or define a default value.'.format(option))
decouple.UndefinedValueError: EXCLUDED_IMAGES not found. Declare it as envvar or define a default value.
```
The user shouldn't be required to use/define EXCLUDED_IMAGES

File: app.py [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+from decouple import config
+import ast
+
+
+class MetricsFetcher:
+    def __init__(self):
+        print('Starting...')
+
+    def store_array_in_yaml(self):
+        excluded_images = ast.literal_eval(
```
Comment (mirii1994) [inline app.py:12]: Why not use os.environ?

File: README.md (hunk 12 above)
Comment (Doron-Bargo) [reply README.md:40]: @resdenia - what about this issue?

File: app.py [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+            os.environ['EXCLUDED_IMAGES'])
```
Comment (Doron-Bargo) [inline app.py:12]: The parameters' table in Readme missing the EXCLUED_IMAGES parameter
Please add to the table, and explain on how to use this environment variable

File: app.py [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+            collection_interval = os.environ['COLLECTION_INTERVAL']
+        else:
+            collection_interval = '30s'
+
+        if 'DOCKER_ENDPOINT' in os.environ:
+            endpoint = os.environ['DOCKER_ENDPOINT']
+        else:
+            endpoint = 'unix:///var/run/docker.sock'
+
+        if 'DOCKER_ENDPOINT' in os.environ:
```
Comment (Doron-Bargo) [inline app.py:38]: Should be TIMEOUT and not DOCKER_POINT

File: app.py [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+            timeout = os.environ['TIMEOUT']
+        else:
+            timeout = '5s'
```
Comment (Doron-Bargo) [inline app.py:41]: In the Readme you wrote that the default value for timeout is 30s but here is 5s . Which one is the correct one?
Please fix

File: app.py [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+
+        with open('./config.yaml') as file:
+            try:
+                config_data = yaml.safe_load(file)
+                if not 'excluded_images' in config_data:
```
Comment (Doron-Bargo) [inline app.py:46]: Why do you have this condition? 
If i understand correct you update the collection_interval, endpoint and timeout only if customer decide not to use excluded_images?
What happen if he decide that he want to exclude images and change those parameters

File: app.py [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+    def store_array_in_yaml(self):
+        excluded_images = ast.literal_eval(
+            os.environ['EXCLUDED_IMAGES'])
+        if  len(excluded_images) > 2:
+
+            with open('./config.yaml') as file:
+                try:
+                    # print(yaml.safe_load(file))
+                    config_data = yaml.safe_load(file)
+                    if not 'excluded_images' in config_data:
```
Comment (Doron-Bargo) [inline app.py:19]: Why you need this condition ?
If customer decide to use excluded_images and for some reason the config already have it shouldnt we overwrite the config with the new parameter?

File: app.py [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+                        config_data['receivers']['docker_stats']['excluded_images'] = excluded_images
+                except yaml.YAMLError as exc:
+                    print(exc)
+
+            with open('./config.yaml', 'w') as file:
+                yaml.dump(config_data, file)
+
+    def default_params(self):
+        if 'COLLECTION_INTERVAL' in os.environ:
```
Comment (Doron-Bargo) [inline app.py:28]: better use os.getenv(key, default = None) instead the if condition

File: app.py [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+            collection_interval = os.environ['COLLECTION_INTERVAL']
+        else:
+            collection_interval = '30s'
+
+        if 'DOCKER_ENDPOINT' in os.environ:
```
Comment (Doron-Bargo) [inline app.py:33]: better use os.getenv(key, default = None) instead the if condition

File: app.py (hunk 17 above)
Comment (Doron-Bargo) [inline app.py:38]: better use os.getenv(key, default = None) instead the if condition

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]
Please see my comments

**Pull request closed without merging**